src/Services/Dominion/Actions/InvadeActionService.php

```diff
@@ -72,7 +72,9 @@ public function __construct(
      */
     public function invade(Dominion $dominion, Dominion $target, array $units): array
     {
-        DB::transaction(function () use ($dominion, $target, $units) {
+        $tempLogObject = [];
+
+        DB::transaction(function () use ($dominion, $target, $units, $tempLogObject) {
```

Comment:
You might want to pass $tempLogObject as a reference here, by using &$tempLogObject. So changes you make to it persist outside of the transaction lambda (I see you dd()-ing below this function)

Reply:
Yeah thank you